crates/biome_js_analyze/src/lint/complexity/no_banned_types.rs

```diff
@@ -149,11 +152,20 @@ declare_lint_rule! {
     /// type notNull<T> = T & {};
     /// ```
     ///
+    /// [^1]: For those curious, this occurs as a natural consequence of TypeScript's type system being _structural_ instead of nominal.
+    /// `{}`, instead of requiring compatible types have _exactly_ 0 properties, simply requires they have _0 or more_ properties.
+    /// This is the exact same mechanism that allows passing `{ foo: number, bar: string }`
+    /// to a function expecting `{ bar: string }` (albeit much more surprising).
+    /// [^2]: In this case, you'd write `declare const myUniqueInternalSymbol: unique symbol` somewhere in the same file.
     pub NoBannedTypes {
         version: "1.0.0",
         name: "noBannedTypes",
         language: "ts",
-        sources: &[RuleSource::EslintTypeScript("ban-types").same()],
+        sources: &[
+            RuleSource::EslintTypeScript("no-empty-object-type").same()
+            RuleSource::EslintTypeScript("no-wrapper-object-types").same()
+            RuleSource::EslintTypeScript("no-unsafe-function-type").same()
+        ],
```

Comment:
1. invalid syntax
2. this changes the behavior of `biome migrate`, so it needs a changeset
3. These ones should be inspired, not same
4. Keep the ban-types rule source, and keep it as same (not inspired) since its a direct port of that rule originally